src/lib/flexbox/api/class.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  Renderer,
+  OnChanges,
+  SimpleChanges,
+  IterableDiffers,
+  KeyValueDiffers
+} from '@angular/core';
+import {NgClass} from '@angular/common';
+
+import {BreakPointRegistry} from './../../media-query/breakpoints/break-point-registry';
+import {ResponsiveActivation} from './../responsive/responsive-activation';
+import {BaseFxDirective} from './base';
+import {MediaChange} from '../../media-query/media-change';
+import {MediaMonitor} from '../../media-query/media-monitor';
+
+/** NgClass allowed inputs **/
+export type NgClassType = string|string[]|Set<string>|{[klass: string]: any};
+
+/**
+ * Directive to add responsive support for ngClass.
+ */
+@Directive({
+  selector: `
+    [class.xs],
+    [class.gt-xs],
+    [class.sm],
+    [class.gt-sm],
+    [class.md],
+    [class.gt-md],
+    [class.lg],
+    [class.gt-lg],
+    [class.xl]
+  `
+})
+export class ClassDirective extends NgClass implements OnInit, OnChanges, OnDestroy {
+
+  @Input('class.xs')    set classXs(val: NgClassType) {
+    this._base.cacheInput('classXs', val);
+  }
+
+  @Input('class.gt-xs') set classGtXs(val: NgClassType) {
```

Comment:
[Google JavaScript Style Guide:](https://google.github.io/styleguide/jsguide.html#formatting-horizontal-alignment) `This practice is permitted, but it is generally discouraged by Google Style`

It makes modification a pain when you have to realign everything assuming it doesn't end up auto-formatted out after a few revisions.